lib/linked_list.rb

```diff
@@ -7,39 +8,79 @@ def initialize
     @head = nil
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
-  def add_first(data)
-
+  # Time complexity - O(1)
+  # Space complexity - O(n)
+  def add_first(value)
+    node = Node.new(value, @head)
+    @head = node
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity - O(1)
+  # Space complexity - O(n)
   def get_first
-
+    return @head.data if @head != nil
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity - O(n)
+  # Space complexity - O(n)
   def length
```

Comment:
Why do you think the space complexity is O(n)?  You're not creating any new lists?